thunder/core/symbol.py

```diff
@@ -526,9 +527,16 @@ def __eq__(self, other):
         return (self.sym, self._var_args, self._var_output) == (other.sym, other._var_args, other._var_output)
 
     def rhs(self):
-        return BoundSymbolRHS(
-            self.sym, tuple(tree_flatten(self._var_args)[0]), tuple(tree_flatten(self._var_kwargs)[0])
-        )
+        def make_hashable(x: Any) -> tuple | FrozenDict:
+            if isinstance(x, dict):
+                return FrozenDict(x)
+            if isinstance(x, list):
+                return tuple(x)
+            return x
+
+        hashable_args = tuple(map(make_hashable, self._var_args))
+
+        return BoundSymbolRHS(self.sym, hashable_args, FrozenDict(self._var_kwargs))
```

Comment:
It's better to convert the values of the dictionary to hashable objects before passing them to FrozenDict.